src/receive_workspace_changes.rs

```diff
@@ -0,0 +1,71 @@
+use std::collections::HashMap;
+use std::fmt::Display;
+use std::sync::Arc;
+use std::time::Instant;
+use ropey::Rope;
+
+use tokio::sync::RwLock as ARwLock;
+use tracing::{info};
+
+use crate::global_context;
+use crate::telemetry;
+
+
+
+#[derive(Debug)]
+pub struct Document {
```

Comment:
why do u move Document here?

Reply:
bc Document is not being used anywhere else but in receive_workspace_changes